arch/riscv/core/isr.S

```diff
@@ -218,7 +426,17 @@ skip_store_fp_caller_saved:
 	la ra, no_reschedule
 	tail _Fault
 
-is_syscall:
+is_kernel_syscall:
```

Comment:
I'm a bit confused on terminology here. What is the use-case for "kernel_syscall"?

Invoking a software trap to invoke a system call when the execution context is already in supervisor mode is undefined behavior in Zephyr. You may safely assume that if a system call trap was generated, it came from user mode.

Unless the Risc V arch implementation is using `ecall` for something else already?

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/master/arch/riscv/core/swap.S#L27

As you can see RISCV isn't like ARM. Here ECALL is used for swap, even if we are already in supervisor mode (it's on master branch)
Before my patch, user wasn't supported then they called that "is_syscall". But now I check at the begining of syscall if it come from user or kernel because each execution is different.
In this patchset, I use Kernel syscall for 2 additional task:
- Usermode enter, because to change privilege level we must be in ECALL.
- Manage return of user syscall function, because it's running in kernel privilege (and priv stack of course) outside of ECALL to be preemptive.